### PR TITLE
Replacing ResourceTestRule with ResourceExtension in documentation

### PR DESCRIPTION
**Pull request by WilliamHerbert**
###### Problem:
Dropwizard 2.0 deprecated ResourceTestRule in favor of ResourceExtension. The documentation providing examples of testing resources was not updated to reflect this.

###### Solution:
ResourceTestRule has been removed from the documentation in favor of ResourceExtension.

This work was completed in `auth.rst` and `forms.rst` but not `testing.rst` as that will be handled in #3062 

###### Result:
Documentation is now more up-to-date with reality. :)

### CODE REVIEW

File: docs/source/manual/auth.rst [hunk 1]
```diff
@@ -280,12 +280,12 @@ Add this dependency into your ``pom.xml`` file:
...
 -------------
 
-When you build your ``ResourceTestRule``, add the ``GrizzlyWebTestContainerFactory`` line.
+When you build your ``ResourceExtension``, add the ``GrizzlyWebTestContainerFactory`` line.
 
 .. code-block:: java
 
     @Rule
-    public ResourceTestRule rule = ResourceTestRule
+    public ResourceExtension rule = ResourceExtension
```
Comment (joschi) [inline docs/source/manual/auth.rst]: Could we rename this field to `extension` or `resourceExtension`?

**Comment by WilliamHerbert** [comment]
I standardized to `resourceExtension`, per suggestion.

File: docs/source/manual/auth.rst [hunk 2]
```diff
@@ -280,12 +280,12 @@ Add this dependency into your ``pom.xml`` file:
 OAuth Example
 -------------
 
-When you build your ``ResourceTestRule``, add the ``GrizzlyWebTestContainerFactory`` line.
+When you build your ``ResourceExtension``, add the ``GrizzlyWebTestContainerFactory`` line.
 
 .. code-block:: java
 
     @Rule
```
Comment (jplock) [inline docs/source/manual/auth.rst]: Do these @Rule and @ClassRule annotations need to be replaced?

Comment (WilliamHerbert) [reply docs/source/manual/auth.rst]: Good catch.  I cleaned up the old annotations and added the ExtendsWith annotation to the test class.

**Comment by cobbce** [comment]
Is this documentation update available in the current documentation on dropwizard.io? I'm struggling to keep track of what version the documentation is given the Getting Started guide says "First, add a dropwizard.version property to your POM with the current version of Dropwizard (which is 2.0.1):" but it appears that the documentation itself is not yet updated?

**Comment by joschi** [comment]
@cobbce The changes of this PR are published on https://www.dropwizard.io/.

You can find the specific pages here:

https://www.dropwizard.io/en/latest/manual/auth.html
https://www.dropwizard.io/en/latest/manual/forms.html

**Pull request merged**